src/lineup/internal/cmds/filter.ts

```diff
@@ -0,0 +1,126 @@
+
+/**
+ * Serialize RegExp objects from LineUp string columns as plain object
+ * that can be stored in the provenance graph
+ */
+interface IRegExpFilter {
+  /**
+   * RegExp as string
+   */
+  value: ILineUpStringFilterValue;
+  /**
+   * Flag to indicate the value should be restored as RegExp
+   */
+  isRegExp: boolean;
+}
+
+/**
+ * This interface combines the `IStringFilter` from `StringColumn`
+ * and `ICategoricalFilter` from `ICategoricalColumn`.
+ */
+interface ILineUpStringFilter {
+  /**
+   * Filter value
+   */
+  filter: ILineUpStringFilterValue | RegExp;
+
+  /**
+   * Filter for missing values
+   */
+  filterMissing: boolean;
+}
+
+/**
+ * Similar to the `ILineUpStringFilter`, but the RegExp is replaced with `IRegExpFilter`
+ */
+interface ISerializableLineUpFilter {
+  /**
+   * Filter value
+   * Note that the RegExp is replaced with IRegExpFilter (compared to the `ILineUpStringFilter` interface)
+   */
+  filter: ILineUpStringFilterValue | IRegExpFilter;
+
+  /**
+   * Filter for missing values
+   */
+  filterMissing: boolean;
+}
+
+/**
+ *  Filter value
+ */
+type ILineUpStringFilterValue = string[] | string | null;
+
+
+/**
+ * This type guard checks if a given parameter has the `filter` property.
+ * Necessary since number columns filter has properties `min`, `max` and no filter property,
+ * @interal
+ *
+ * @param filter Any value that could be a filter
+ */
+export function isSerializedFilter(filter: any): ISerializableLineUpFilter {
```

Comment:
```suggestion
export function isSerializedFilter(filter: any): filter is ISerializableLineUpFilter {
```

I would expect that this is a type guard function. And the `return` statement indicates a boolean return value. Please revise.

Reply:
The function checks if the filter is `ISerializableLineUpFilter | ILineUpStringFilter`, so i added a typeguard that includes both.

Reply:
That fix was only part of the solution. The reason why you had to add `ILineUpStringFilter` to everything was that we were using the function `isSerializedFilter` in `recordPropertyChange` which is incorrect! Instead we must check if the provided filter is a valid `ILineUpStringFilter`. I refactored the code and extracted + tested the type guards.